src/hypothesis/internal/charmap.py

```diff
@@ -50,25 +54,25 @@ def charmap():
                     rs[-1][-1] += 1
                 else:
                     rs.append([i, i])
-            # We explicitly set the mtime to an arbitrary value so as to get
-            # a stable format for our charmap.
             data = sorted(
                 (k, tuple((map(tuple, v))))
                 for k, v in tmp_charmap.items())
 
-            # Write the Unicode table atomically
-            fd, tmpfile = tempfile.mkstemp(dir=tmpdir())
-            os.close(fd)
-            with GzipFile(tmpfile, 'wb', mtime=1) as o:
-                o.write(pickle.dumps(data, pickle.HIGHEST_PROTOCOL))
+            _charmap = dict(data)
+
             try:
+                # Cache the result of this operation, to speed up startup
+                # Write the Unicode table atomically
+                fd, tmpfile = tempfile.mkstemp(dir=tmpdir())
+                os.close(fd)
+                # We explicitly set the mtime to an arbitrary value so as to
+                # get a stable format for our charmap.
+                with GzipFile(tmpfile, 'wb', mtime=1) as o:
+                    o.write(pickle.dumps(data, pickle.HIGHEST_PROTOCOL))
                 os.rename(tmpfile, f)
-            except FileExistsError:  # pragma: no cover
-                # This exception is only raised on Windows, and coverage is
-                # measured on Linux.
+
+            except Exception:
```

Comment:
Add a magic comment here: `#pragma: no cover`, as the coverage run will never have this operation fail.  That's the only one you need!